torch/_refs/__init__.py

```diff
@@ -2846,6 +2848,55 @@ def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLi
     return where(mask, prims.to_dtype(value, a.dtype), a)
 
 
+def allclose(
+    a: TensorLikeType,
+    b: TensorLikeType,
+    rtol: Optional[Union[bool, int, float]] = None,
```

Comment:
> Why this rather than Optional[NumberType]?

```py
NumberType = Union[bool, int, float, complex]
```

it's to remove `complex`.

```py
>>> import torch
>>> x = torch.tensor([1])
>>> torch.allclose(x,x,1+1j)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: allclose(): argument 'rtol' (position 3) must be float, not complex

>>> torch.allclose(x,x,2)
True
>>> torch.allclose(x,x,True)
True
>>> torch.allclose(x,x,2.2)
True
```